executor/analyze.go

```diff
@@ -142,10 +142,15 @@ type analyzeTask struct {
 	Columns   []*model.ColumnInfo
 	PKInfo    *model.ColumnInfo
 	src       Executor
+	pushdown  bool
 }
 
 func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultCh chan<- statistics.AnalyzeResult) {
 	for task := range taskCh {
+		if task.pushdown {
+			resultCh <- e.handleAnalyzePushDown(task)
```

Comment:
I think we should move this under the `idxTask` case at line 158.
And the name should be the same pattern as `analyzeIndex`, `analyzeColumns`, like `analyzeIndexPushdown`.